src/components/Dashboard/Dashboard.js

```diff
@@ -42,101 +51,93 @@ const dashboardMap = {
     [PRINT_LAYOUT]: <PrintLayoutDashboard />,
 }
 
-export class Dashboard extends Component {
-    setDashboard = () => {
-        if (this.props.dashboardsLoaded) {
-            const id = this.props.match.params.dashboardId || null || null
-
-            this.props.selectDashboard(id)
-
-            this.setHeaderbarVisibility()
-        }
-    }
-
-    setHeaderbarVisibility = () => {
-        const header = document.getElementsByTagName('header')[0]
-        if (isPrintMode(this.props.mode)) {
-            header.classList.add('hidden')
-        } else {
-            header.classList.remove('hidden')
+export const Dashboard = ({
+    id,
+    mode,
+    dashboardsLoaded,
+    dashboardsIsEmpty,
+    routeId,
+    selectDashboard,
+    setWindowHeight,
+}) => {
+    useEffect(() => {
+        setHeaderbarVisibility(mode)
+    }, [mode])
+
+    useEffect(() => {
+        if (dashboardsLoaded && !dashboardsIsEmpty) {
+            selectDashboard(routeId)
         }
-    }
+    }, [dashboardsLoaded, dashboardsIsEmpty, routeId, mode])
 
-    componentDidMount() {
+    useEffect(() => {
+        // TODO does this need to be cleared?
         window.onresize = debounce(
-            () => this.props.setWindowHeight(window.innerHeight),
+            () => setWindowHeight(window.innerHeight),
```

Comment:
@jenniferarnesen yeah this effect should clean up the event handler on unmount, so something like this:

```js
useEffect(() => {
  const onResize = debounce(() => setWindowHeight(window.innerHeight))
  window.addEventListener('resize', onResize);
  return () => {
    window.removeEventListener('resize', onResize)
  }
}
```